compiler/rustc_middle/src/ty/diagnostics.rs

```diff
@@ -54,7 +54,7 @@ impl<'tcx> TyS<'tcx> {
     /// ADTs with no type arguments.
     pub fn is_simple_text(&self) -> bool {
         match self.kind() {
-            Adt(_, substs) => substs.types().next().is_none(),
+            Adt(_, substs) => substs.types().next().is_none() && substs.consts().next().is_none(),
```

Comment:
maybe use `non_erasable_generics`? Not that we'll be adding more stuff soon, but the point of `is_simple_text` is basically "check that there are only lifetimes in any generic list"